### PR TITLE
chore(e2e): switch to new transparent proxy engine

### PR DESCRIPTION
**Pull request by lukidzi**
Use by default our new TP engine in E2E tests.

### Checklist prior to review
- [X] Link to docs PR or issue --
- [X] Link to UI issue or PR --
- [X] Is the [issue worked on linked][1]? -- fix: https://github.com/kumahq/kuma/issues/4442
- [X] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [X] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Unit Tests --
- [X] E2E Tests --
- [X] Manual Universal Tests --
- [X] Manual Kubernetes Tests --
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [X] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: test/e2e_env/universal/gateway/gateway.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ func Gateway() {
 	})
 
 	E2EAfterAll(func() {
+		Expect(env.Cluster.DeleteApp("gateway-client")).To(Succeed())
```
Comment (michaelbeaumont) [inline test/e2e_env/universal/gateway/gateway.go:37]: This can/should really be fixed in a separate PR to `master` assuming it's unrelated

File: test/framework/interface.go [hunk 2]
```diff
@@ -88,29 +90,30 @@ type appDeploymentOptions struct {
...
+	kumactlFlow                  bool
+	concurrency                  int
+	omitDataplane                bool
+	proxyOnly                    bool
+	serviceProbe                 bool
+	reachableServices            []string
+	appendDataplaneConfig        string
+	boundToContainerIp           bool
+	serviceAddress               string
+	experimantalTransparentProxy bool
```
Comment (michaelbeaumont) [inline test/framework/interface.go:116]:
```suggestion
	experimentalTransparentProxy bool
```

File: test/framework/universal_app.go [hunk 3]
```diff
@@ -458,14 +458,25 @@ func (s *UniversalApp) CreateDP(
 	s.dpApp = ssh.NewApp(s.containerName, s.verbose, s.ports[sshPort], nil, args)
 }
 
-func (s *UniversalApp) setupTransparent(cpIp string, builtindns bool) {
+func (s *UniversalApp) setupTransparent(cpIp string, builtindns bool, experimantalTransparentProxy bool) {
```
Comment (michaelbeaumont) [inline test/framework/universal_app.go:461]:
```suggestion
func (s *UniversalApp) setupTransparent(cpIp string, builtindns bool, experimentalTransparentProxy bool) {
```

File: go.mod [hunk 4]
```diff
@@ -27,7 +27,7 @@ require (
 	github.com/gruntwork-io/terratest v0.41.4
 	github.com/hoisie/mustache v0.0.0-20160804235033-6375acf62c69
 	github.com/kelseyhightower/envconfig v1.4.0
-	github.com/kumahq/kuma-net v0.8.9
+	github.com/kumahq/kuma-net v0.8.10
```
Comment (michaelbeaumont) [inline go.mod:30]: also separate PR assuming the upgrade works without the rest of PR

**Review by michaelbeaumont — APPROVED** [review]
LGTM otherwise but not a `kuma-net` expert

File: test/e2e_env/universal/gateway/gateway.go (hunk 1 above)
Comment (lukidzi) [reply test/e2e_env/universal/gateway/gateway.go:37]: https://github.com/kumahq/kuma/pull/5513/files

File: go.mod (hunk 4 above)
Comment (lukidzi) [reply go.mod:30]: https://github.com/kumahq/kuma/pull/5513/files

**Review by jakubdyszkiewicz — APPROVED** [review]

**Review by slonka — APPROVED** [review]
